src/sinks/amqp/sink.rs

```diff
@@ -147,7 +147,7 @@ impl AmqpSink {
             })
             .into_driver(service);
 
-        sink.run().await
+        sink.run(Some("amqp_0_9_1".into())).await
```

Comment:
Not to rock the boat .... and I'm not an expert on this sink framework code by any means so I'm not sure if this is a realistic idea but, I'm wondering if it might be a bit more organized to set the protocol in a function rather than have it be a parameter to the run function. Like:


```Rust

        let sink = input
            .filter_map(|event| std::future::ready(self.make_amqp_event(event)))
            .request_builder(None, request_builder)
            .filter_map(|request| async move {
                match request {
                    Err(e) => {
                        error!("Failed to build AMQP request: {:?}.", e);
                        None
                    }
                    Ok(req) => Some(req),
                }
            })
            .into_driver(service)
            .protocol(Some("amqp_0_9_1".into());
            
        sunk.run().await;
```

Just something to consider.

Reply:
That's a good idea, instead of using an `Option` that is `None` in most of the places.